### PR TITLE
fix(windows): nsis failed to resolve resources with `$` in their name, closes #9657

### LINKED ISSUE #9657: [bug] Resources with `$` in their name make build step fail on Windows
### Describe the bug

I'm working on an app that utilizes JNI to call Java. In my Java class, I have inner classes that should be named OuterClass$InnerClass.class after compilation. However, these inner class files cannot be found during the build step. It appears that the $ symbol is being improperly escaped with another $, causing the build step to fail.

### Reproduction

1. Create a Next.js Tauri app using Next.js.
2. Add a resource with $ in its name to the tauri.conf.json.
3. Build the app.

### Expected behavior

The build step should finish with included resources.

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22621 X64
    ✔ WebView2: 124.0.2478.67
    ✔ MSVC: 
        - Visual Studio Community 2019
        - Visual Studio Community 2022
    ✔ rustc: 1.77.1 (7cf61ebde 2024-03-27)
    ✔ cargo: 1.77.1 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.12.0
    - pnpm: 8.15.6
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 1.5.4
    - tauri-build [RUST]: 1.5.1
    - wry [RUST]: 0.24.7
    - tao [RUST]: 0.16.7
    - tauri-cli [RUST]: 1.5.11
    - @tauri-apps/api [NPM]: 1.5.3 (outdated, latest: 1.5.4)
    - @tauri-apps/cli [NPM]: 1.5.11 (outdated, latest: 1.5.12)

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../out
    - devPath: http://localhost:3000/
    - framework: React (Next.js)
    - bundler: Webpack
```

### Stack trace
The actual file is named `WebFXScan$CImageInfo.class`
```text
File: "C:\Users\enkhj\projects\pms-mirai-app\src-tauri\java\adview\webfxscan\WebFXScan$$CImageInfo.class" -> no files found.
Usage: File [/nonfatal] [/a] ([/r] [/x filespec [...]] filespec [...] |
   /oname=outfile one_file_only)
Error in script "C:\Users\enkhj\projects\pms-mirai-app\src-tauri\target\release\nsis\x64\installer.nsi" on line 563 -- aborting creation process
       Error failed to bundle project: `The system cannot find the file specified. (os error 2)`
```

### Additional context

It works fine on MacOS

### PR DESCRIPTION
**Pull request by enkhjile**
Registering a global escape function that replaces `$` with `$$` in Handlebars causes the build step to fail when resources contain `$` in their names, because NSIS commands like `File /a` cannot resolve the escaped paths. #9657

I've added a helper that unescapes `$$` back to `$` and modified the corresponding resource expression in `installer.nsi` to use this helper.

If there are better solutions, I'd be happy to implement them.

### CODE REVIEW

File: tooling/bundler/src/bundle/windows/templates/installer.nsi [hunk 1]
```diff
@@ -543,15 +543,15 @@ Section Install
 
   ; Copy resources
   {{#each resources_dirs}}
-    CreateDirectory "$INSTDIR\\{{this}}"
+    CreateDirectory "$INSTDIR\\{{unescape-dollar-sign this}}"
   {{/each}}
   {{#each resources}}
-    File /a "/oname={{this.[1]}}" "{{@key}}"
+    File /a "/oname={{unescape-dollar-sign this.[1]}}" "{{unescape-dollar-sign @key}}"
```
Comment (amrbashir) [inline tooling/bundler/src/bundle/windows/templates/installer.nsi]: no need to unescape in `/oname` portion

File: tooling/bundler/src/bundle/windows/templates/installer.nsi [hunk 2]
```diff
@@ -543,15 +543,15 @@ Section Install
...
   {{/each}}
 
   ; Copy external binaries
   {{#each binaries}}
-    File /a "/oname={{this}}" "{{@key}}"
+    File /a "/oname={{unescape-dollar-sign this}}" "{{unescape-dollar-sign @key}}"
```
Comment (amrbashir) [inline tooling/bundler/src/bundle/windows/templates/installer.nsi]: no need to unescape in `/oname` portion

**Comment by amrbashir** [comment]
Thanks for your contributions. I did a bit of experiments and seems like only the input to `File`, needs unescaped values so please revert changes to other places, also could you run `cargo fmt` inside `tooling/bundler`?

**Comment by enkhjile** [comment]
@amrbashir 
Thanks for your feedback. I have reverted the changes to the `installer.nsi`, except for the `File` sections, and have run `cargo fmt`.

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thank you, just a couple of more to go

File: tooling/bundler/src/bundle/windows/templates/installer.nsi [hunk 3]
```diff
@@ -546,12 +546,12 @@ Section Install
...
```
Comment (amrbashir) [inline tooling/bundler/src/bundle/windows/templates/installer.nsi]:
```suggestion
    File /a "/oname={{this}}" "{{unescape-dollar-sign @key}}"
```

File: tooling/bundler/src/bundle/windows/templates/installer.nsi [hunk 4]
```diff
@@ -546,12 +546,12 @@ Section Install
     CreateDirectory "$INSTDIR\\{{this}}"
   {{/each}}
   {{#each resources}}
-    File /a "/oname={{this.[1]}}" "{{@key}}"
+    File /a "/oname={{unescape-dollar-sign this.[1]}}" "{{unescape-dollar-sign @key}}"
```
Comment (amrbashir) [inline tooling/bundler/src/bundle/windows/templates/installer.nsi]:
```suggestion
    File /a "/oname={{this.[1]}}" "{{unescape-dollar-sign @key}}"
```

**Review by amrbashir — DISMISSED** [review]
Thank you

**Review by amrbashir — APPROVED** [review]

**Pull request merged**